oauth.py

```diff
@@ -58,10 +58,10 @@ def callback(self):
                   'grant_type': 'authorization_code',
                   'redirect_uri': self.get_callback_url()}
         )
-        me = oauth_session.get('me?fields=id,email').json()
+        me = oauth_session.get('me?fields=id,name,email').json()
         return (
             'facebook$' + me['id'],
-            me.get('email').split('@')[0],  # Facebook does not provide
+            me.get('name'),  # Facebook does not provide
```

Comment:
As far as I know, the name was always returned. What FB does not return is a username. The name is not great to use as username, since it can contain spaces, foreign characters, etc.